keel-titus-plugin/src/test/kotlin/com/netflix/spinnaker/keel/titus/ContainerAttributesResolverTests.kt

```diff
@@ -0,0 +1,130 @@
+import com.netflix.spinnaker.keel.api.titus.TitusClusterSpec
+import com.netflix.spinnaker.keel.api.titus.TitusServerGroupSpec
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverService
+import com.netflix.spinnaker.keel.docker.ReferenceProvider
+import com.netflix.spinnaker.keel.test.resource
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.mockk
+import org.springframework.core.env.Environment
+import strikt.api.expect
+import strikt.api.expectThat
+import strikt.assertions.containsExactly
+import strikt.assertions.isEqualTo
+
+internal class ContainerAttributesResolverTests : JUnit5Minutests {
+
+  val accountKey = "titus.account"
+  val subnetKey = "titus.subnet"
+  val eastSubnets = "subnet-east-1, subnet-east-2"
+  val westSubnets = "subnet-west-3, subnet-west-4"
+  val account = "titus"
+  val awsAccountId = "1234"
+
+  val defaults = mockk<DefaultContainerAttributes> {
+    every { getAccountKey() } returns accountKey
+    every { getSubnetKey() } returns subnetKey
+    every { getSubnetValue(account, "east")} returns eastSubnets
+    every { getSubnetValue(account, "west")} returns westSubnets
+  }
+
+  val baseSpec = TitusClusterSpec(
+    moniker = Moniker(app = "keel", stack = "test"),
+    locations = SimpleLocations(
+      account = account,
+      regions = setOf(SimpleRegionSpec("east"), SimpleRegionSpec("west"))
+    ),
+    container = ReferenceProvider("my-artifact"),
+    _defaults = TitusServerGroupSpec(),
+    overrides = emptyMap()
+  )
+
+  val clouddriverService = mockk<CloudDriverService>() {
+    coEvery { getAccountInformation(account, any()) } returns mapOf("awsAccount" to "aws")
+    coEvery { getAccountInformation("aws", any()) } returns mapOf("accountId" to awsAccountId)
+  }
+  val springEnv: Environment = mockk(relaxed = true) {
+    coEvery { getProperty("keel.titus.resolvers.container-attributes.enabled", Boolean::class.java, true) } returns true
+  }
+  data class Fixture(val subject: ContainerAttributesResolver, val spec: TitusClusterSpec) {
+    val resource = resource(
+      kind = TITUS_CLUSTER_V1.kind,
+      spec = spec
+    )
+    val resolved by lazy { subject(resource) }
+  }
+
+
+  fun tests() = rootContext<Fixture> {
+    fixture {
+      Fixture(
+        ContainerAttributesResolver(defaults, clouddriverService, springEnv),
+        baseSpec
+      )
+    }
+
+    test("supports the resource kind") {
+      expectThat(listOf(subject).supporting(resource))
+        .containsExactly(subject)
+    }
+
+    context("spec has no defaults set") {
+      test("account and subnet are set") {
+        validateKeysSet(resolved)
+      }
+    }
+
+    context("account is set") {
+      fixture {
+        Fixture(
+          ContainerAttributesResolver(defaults, clouddriverService, springEnv),
+          baseSpec.copy(_defaults = baseSpec.defaults.copy(containerAttributes = mapOf(accountKey to awsAccountId)))
```

Comment:
I would recommend using a different account id here (or adding another test with a different account id), to catch the bug where the user specifies a custom account id, and the resolver clobbers it with the default.

Reply:
ok i'll do that, good point

Reply:
oh wait i do this in the following test!